### DISCUSSION TITLE
How to import?

### DISCUSSION THREAD

**Discussion by explosion-mental** [mitsuhiko/minijinja#509] [Q&A] [answered]
Hi, I get template not found "template.j2" when using `{% import "template.j2" as test %}`. I'm not an expert on Jinja2 syntax, but I want to make some variables and reuse them in my templates.

"template.j2"
```jinja2
{% set primary = "#4455AA" %}
{% set name = "test" %}
```

I'm unsure if I neet to `.add_template` the import file (the file with the variables) or why I get that the file isn't found `Could not render template: template not found: tried to include non-existing template "template.j2"`

**Comment by mitsuhiko** [comment, marked as answer]
Yes, you need to `add_template` that file into the environment. Alternatively you can use a loader to automatically do that. You can find some examples here:

- basic path loader (loads from file system): https://github.com/mitsuhiko/minijinja/blob/main/examples/path-loader/src/main.rs
- example using embedding (bundled) with fallback to path loader in dev: https://github.com/mitsuhiko/minijinja/blob/main/examples/embedding/src/main.rs
